src/assets/php8.4/php8.4.md

```diff
@@ -0,0 +1,226 @@
+
+Instead of ending the property with a semicolon, we can use a code block ({}) to include the property hook.
+These hooks are optional and can be used independently. By excluding one or the other we can make the property read-only or write-only.
+
+### Example
+
+```
+class User
+{
+    public function __construct(private string $first, private string $last) {}
+
+    public string $fullName {
+      get => $this->first . " " . $this->last;
+
+      set (string $value) {
+	      if (strlen($value) === 0) {
+          throw new ValueError("Name must be non-empty");
+        }
+        [$this->first, $this->last] = explode(' ', $value, 2);
+      }
+    }
+}
+
+$user = new User('Alice', 'Hansen')
+$user->fullName = 'Brian Murphy';  // the set hook is called
+echo $user->fullName;  // "Brian Murphy"
+echo $user->first;  // "Brian"
+```
+
+
+## 3. 'new' without parentheses
+
+PHP 8.4 introduces a easier syntax, allowing you to chain methods on newly created objects without parenthesis. Although this is a minor adjustment, it results in cleaner and less verbose code.
+```
+(new MyClass())->getShortName();  // PHP 8.3 and older
+new MyClass()->getShortName();  // PHP 8.4
+```
+
+Besides chaining methods on newly created objects, you can also chain properties, static methods and properties, array access, and even direct invocation of the class. For example:
+```
+new MyClass()::CONSTANT,
+new MyClass()::$staticProperty,
+new MyClass()::staticMethod(),
+new MyClass()->property,
+new MyClass()->method(),
+new MyClass()(),
+new MyClass(['value'])[0],
+```
+
+## 4. Implicitly nullable parameter declarations depreciated
+
+Before PHP 8.4, if a parameter was of type `X`, it could accept a null value without explicitly declaring `X` as nullable. Starting with PHP 8.4, you can no longer declare a null parameter value without clearly stating it as nullable in the type hint; otherwise, a deprecation warning will be triggered. 
+```
+function greetings(string $name = null)  // fires a deprecation warning
+```
+
+To avoid warnings, you must explicitly state that a parameter can be null by using a question mark (?) in the type declaration.
+```
+function greetings(?string $name)
+```
+or,
+
+```
+function greetings(?string $name = null)
+```
+
+
+## 5. New multibyte functions
+
+A multibyte string is a sequence of characters where each character can use more than one byte of storage. This is common in languages with complex or non-Latin scripts, such as Japanese or Chinese. There are several multibyte functions in PHP such as mb_strlen(), mb_substr(), mb_strtolower(), mb_strpos(), etc. But some of the functions like trim(), ltrim(), rtrim(), ucfirst(), lcfirst() etc. lack direct multibyte equivalents.
```

Comment:
maybe also use the same styling as [above](https://github.com/JankariTech/blog/pull/157/files#r1741497617) for the functions listed here?